portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+
+/** Servlet that returns comment information. */
+@WebServlet("/api/comments")
+@MultipartConfig
+public class CommentServlet extends HttpServlet {
+  private static Gson gson = new Gson();
+  private static ArrayList<Comment> COMMENTS = new ArrayList<Comment>();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = gson.toJson(COMMENTS);
+    response.setContentType("application/json");
+    response.getWriter().print(json);
+  }
+
+  /**
+   * This route expects multipart form data, not url-encoded form data.
+   */
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String username = readPartToString(req.getPart("username"));
```

Comment:
use getParameter instead of getPart to get your parameter as a string. You shouldn't have to parse bytes unless you're handling file uploads.

Reply:
`getParameter` did not work, because the form data is being sent as `multipart/form-data` (which is what happens when you send a `FormData` object using `fetch()`). 

I could work around that on the client side, but this seemed like a more robust solution because it also makes it easy for me to do server-side validation (in a future PR I take advantage of this by truncating comment text at 50000 bytes, which I could not do if I was not handling the bytes directly. If I did not do this, it would make it pretty trivial for someone to DDoS my app).

Reply:
TIL. Resolving.